libraries/common-arm/src/sd/sd_interface.rs

```diff
@@ -0,0 +1,160 @@
+use core::marker::PhantomData;
+
+use defmt::{info, warn};
+
+// use atsamd_hal as hal;
+use embedded_sdmmc as sd;
+use atsamd_hal::gpio::{Pin, PA17, PA19, PA16, PA18, Output, PushPull, Alternate, C};
+use atsamd_hal::sercom::{spi, IoSet1, Sercom1};
+use atsamd_hal::clock::{Sercom1CoreClock};
+use atsamd_hal::pac;
+
+pub struct TimeSink {
```

Comment:
This doesn't need to be public

Reply:
TimeSink has to be public otherwise the result is an error[E0446]: private type `TimeSink` in public interface.